file/file_prefetch_buffer.cc

```diff
@@ -853,7 +854,7 @@ bool FilePrefetchBuffer::TryReadFromCacheUntracked(
     } else {
       return false;
     }
-  } else if (!for_compaction) {
+  } else {
```

Comment:
I'd prefer to add back the check `!for_compaction` here, as the stats are mainly for visibility into the effectiveness of prefetching for user reads.

Reply:
Sounds good. I updated the code and tests, and added another comment for future readers